### ISSUE TITLE
Enhancement: [array-type] Detect `Readonly<string[]>`

### ISSUE THREAD

**Issue by neelance** [typescript-eslint/typescript-eslint#7755]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/array-type

### Description

The rule should also turn `Readonly<string[]>` into `readonly string[]`.

### Fail

```typescript
const y: Readonly<string[]> = ['a', 'b'];
```

### Pass

```typescript
const y: readonly string[] = ['a', 'b'];
```

**Comment by bradzacher** [comment]
Is this a common pattern you've seen around your codebase?

**Comment by JoshuaKGoldberg** [comment]
Enforcing `Readonly<...>` vs. `readonly ...` is covered by https://github.com/eslint-functional/eslint-plugin-functional/blob/e4bd4ad79502ff77ae26e703e1761ab80a46868b/docs/rules/prefer-immutable-types.md and https://github.com/eslint-functional/eslint-plugin-functional/blob/e4bd4ad79502ff77ae26e703e1761ab80a46868b/docs/rules/type-declaration-immutability.md (previously, https://github.com/eslint-functional/eslint-plugin-functional/blob/e4bd4ad79502ff77ae26e703e1761ab80a46868b/docs/rules/immutable-data.md). That's a separate concern from the Array type styles.

If folks have already started adding `Readonly<...>` to types and are enforcing immutability on certain types, then I bet https://github.com/eslint-functional/eslint-plugin-functional and/or https://github.com/jhusain/eslint-plugin-immutable are plugins you all might find useful (if you haven't already started using them).

Thanks for filing!

**Comment by Josh-Cena** [comment]
@JoshuaKGoldberg To clarify: this issue asks for the rule to check that readonly array types are declared _consistently_, not that they _are_ marked as readonly. Do those rules cover this use case?

**Comment by JoshuaKGoldberg** [comment]
Yeah I think this rule shouldn't cover that case because `Readonly<...>` isn't specific to arrays, and users often end up writing their own equivalents ([`ReadonlyDeep`](https://github.com/sindresorhus/type-fest/blob/010dfdda142fee47460419f93d0ed174533adc2d/source/readonly-deep.d.ts), etc.). Trying to expand this rule's purview to the assorted wrapper types folks use would make its area of responsibilities more than just array types.

**Comment by Josh-Cena** [comment]
This rule already checks between `readonly T[]` and `ReadonlyArray<T>`. It's not unreasonable to me to expect `Readonly<T[]>` to work. But maybe this would be confusing to some users, I don't know.

**Comment by bradzacher** [comment]
I would say that we can easily include this case because it's syntactically simple to detect the case of `Readonly<T[]>` - the developer intent was clear - they wanted a readonly array and there are true constructs for that - so we can improve codebase consistency by reporting and suggesting the configured style instead.

I'm in favour of this, personally!

**Comment by JoshuaKGoldberg** [comment]
Well, I guess I'm being outvoted on this one 😄

**Comment by threema-lenny** [comment]
> Is this a common pattern you've seen around your codebase?

I just stumbled upon this in one of our projects and was very confused that it's not caught by the `array-type` rule. Can confirm that it happens. :sweat_smile:
